specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1136 @@
+The following table shows the supported permutations of verbs and targets:
+
+|Verb         | Path | Filename | PackageFamilyName | PackageFullName | file:  | http(s): | ms-uup: | PackageSet |
+|-------------|:----:|:--------:|:-----------------:|:---------------:|:------:|:--------:|:-------:|:----------:|
+|IsReady      |  X   |    X     |       OS/WAS      |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|EnsureIsReady|  X   |    X     |         X         |        X        |   X    |    X     |  WAS    |    WAS     |
+|Add          | WAS  |    X     |         X         |        X        | OS/WAS |  OS/WAS  | OS/WAS  |    WAS     |
+|Stage        | WAS  |    X     |         X         |        X        | OS/WAS |  OS/WAS  | OS/WAS  |    WAS     |
+|Register     | WAS  |  OS/WAS  |       OS/WAS      |      OS/WAS     | OS/WAS |    X     | OS/WAS  |    WAS     |
+|Remove       |  X   |    X     |        WAS        |      OS/WAS     |   X    |    X     | OS/WAS  |    WAS     |
+|Repair       |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Reset        |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Provision    |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+|Deprovision  |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+
+Legend:
+
+* OS = Supported by Windows (OS) APIs in the Windows.Management.Deployment.PackageManager namespace.
+* WAS = Supported by Windows App SDK APIs in the
+  Microsoft.Windows.Management.Deployment.PackageDeploymentManager namespace.
+* X = Not supported
+
+## 3.2. Is*Ready()
+
+Is*Ready() methods determine if the target is installed and ready for use. Reasons why a package is
+not ready can include:
+
+* The package is not present on the machine
+* The package is present on the machine but not registered for the user
+* The package is registered for the user but is not in a healthy status e.g. its
+  Package.Status=Tampered
+
+Is*Ready() methods are a quick test to determine if more (costly) work is needed before the target
+can be used.
+
+## 3.3. Ensure*IsReady()
+
+Ensure*IsReady() methods determine if the target is installed and ready for use and, if not, makes
+it so. This can include downloading the target, registering it for the user and remediating a
+package in an unhealthy state.
+
+Thus `EnsurePackageIsReady(pkg, options)` is functionally equivalent to
+
+```c#
+var pdm = PackageDeploymentManager().GetDefault();
+if (!pdm.IsPackageReady(pkg))
+{
+    var result = await pdm.AddPackageAsync(pkg, options);
+}
+endif
+```
+
+### 3.3.1. Why Is*Ready() Given Ensure*IsReady()?
+
+Ensure*IsReady() performs an 'is ready' check and returns if all is ready. There's no efficiency
+reasons to call Is*Ready() before Ensure*IsReady() (in fact, it's less efficient as Is*Ready() would
+occur twice).
```

Comment:
Formatting issue: the * characters make everything italic in between.

Reply:
Ewww. Ahhhh markdown. Thanks. Fixing...